packages/hardhat/contracts/Soundly.sol

```diff
@@ -0,0 +1,75 @@
+pragma solidity >=0.8.0 <0.9.0;
+//SPDX-License-Identifier: MIT
+
+import "@openzeppelin-contracts/contracts/access/AccessControl.sol";
+
+contract Soundly is AccessControl {
+
+    // Standard roles definition
+    // bytes32 public constant DEFAULT_ADMIN_ROLE = 0x00;
+    bytes32 public constant DEFAULT_ADMIN_ROLE = keccak256("DefaultAdminRole");
+    bytes32 public constant ADMIN_ROLE = keccak256("AdminRole");
+    bytes32 public constant ARTIST_ROLE = keccak256("ArtistRole");
+
+    mapping(bytes32 => address) artistMusics; // mapping of filecoin/IPFS music ID to the artist address
+    bytes32[] public allMusics;
+    address[] public allArtists;
+
+    // Struct of artist existing and list of artistMusics ID
+    struct Artist {
+        bool isArtist;
+        bytes32[] musicsID;
+    }
+
+    mapping(address => Artist) artists;
+
+    // these two address parameters would be have admin priviledges
+    constructor(address _admin1, address _admin2) public {
+        _setRoleAdmin(ADMIN_ROLE, _msgSender());
+        _setRoleAdmin(ARTIST_ROLE, _msgSender());
```

Comment:
why are you granting msgSender two roles?

Reply:
The oz access control use two levels:
There is an 'Admin' for every role, and then members with the 'role' itself.
So msgSender is the 'Admin' for the every role defined in the contract i.e the 'admin' role and the 'artist' role